in-progress/8509-prover-coordination/design.md

```diff
@@ -0,0 +1,124 @@
+# Prover Coordination
+
+|                      |                                                             |
+| -------------------- | ----------------------------------------------------------- |
+| Issue                | https://github.com/AztecProtocol/aztec-packages/issues/8509 |
+| Owners               | @just-mitch                                                 |
+| Approvers            | @aminsammara @LHerskind @spalladino                         |
+| Target Approval Date | 2024-09-18                                                  |
+
+
+## Executive Summary
+
+The aztec node will provide an interface for proving marketplaces to submit quotes for proving an epoch.
+
+A proposer will be able to respond to a quote with a signed message, which a prover can submit to L1, along with a prover commitment bond. The prover can subsequently submit the proof to L1, releasing the bond.
+
+This coordination will take place "out of protocol", via an optional p2p topic, with the slight exception/leak that the rollup contract will permit the prover to submit bonds/proofs in a proposer's stead.
```

Comment:
Are there any issues from making these quotes public on the p2p layer? Namely:
- Censorship from other provers, who don't relay competing quotes
- Undercutting quotes by competitors (ie I see a quote for 100, so I send one for 99) by making all quotes public
- Delay in propagation until the quote reaches the proposers

Would it be better to have each sequencer keep a list of endpoints to query, and they hit the prover-nodes directly? Or via one or more relays? In practice, do we expect sequencers to take orders from unknown provers, or only from a whitelist they keep?

Reply:
Censorship: As long as a node has sufficient peers, they *shouldn't* be able to be eclipsed by other provers, since regular full nodes and validators can/should also be forwarding these messages. 

undercutting: I see it as a first price auction, which is known to have issues where the winner under/overpays. I'm okay with that in the immediate term.

delay: this should be mitigated by the proof claim period being multiple L2 slots. Assuming 13 slots for proof claim, and 36 second slots, that should be abundant time.

hitting prover nodes directly: I think this is possible, especially if we see, e.g. provers are unwilling to participate in an open auction. This is also something that could be implemented in the community down the line. If we consider the p2p layer as the public mempool for bids which serves as a baseline, I can imagine more privacy preserving options being created down the line, like a list of servers to query to bids, or a relay, etc.

unknown provers: I think that a proposer can do some validation. a simple one would just be to read the L1 balance of the prover to see if they have sufficient TST to post the bond. Peer scoring could be used here. I'd like to avoid relying on a whitelist for the vanilla/baseline solution if possible, but I could see nodes using one in practice.